easy-connect.cpp

```diff
@@ -0,0 +1,289 @@
+                           ATMEL_SPI_RST, ATMEL_SPI_SLP, ATMEL_SPI_IRQ, ATMEL_I2C_SDA, ATMEL_I2C_SCL);
+
+#elif MBED_CONF_APP_MESH_RADIO_TYPE == MCR20
+#include "NanostackRfPhyMcr20a.h"
+NanostackRfPhyMcr20a rf_phy(MCR20A_SPI_MOSI, MCR20A_SPI_MISO, MCR20A_SPI_SCLK, MCR20A_SPI_CS, MCR20A_SPI_RST, MCR20A_SPI_IRQ);
+
+#elif MBED_CONF_APP_MESH_RADIO_TYPE == SPIRIT1
+#include "NanostackRfPhySpirit1.h"
+NanostackRfPhySpirit1 rf_phy(SPIRIT1_SPI_MOSI, SPIRIT1_SPI_MISO, SPIRIT1_SPI_SCLK,
+                             SPIRIT1_DEV_IRQ, SPIRIT1_DEV_CS, SPIRIT1_DEV_SDN, SPIRIT1_BRD_LED);
+
+#elif MBED_CONF_APP_MESH_RADIO_TYPE == EFR32
+#include "NanostackRfPhyEfr32.h"
+NanostackRfPhyEfr32 rf_phy;
+
+#endif // MBED_CONF_APP_RADIO_TYPE
+#endif // MESH
+
+#if defined (WIFI)
+#define WIFI_SSID_MAX_LEN      32
+#define WIFI_PASSWORD_MAX_LEN  64
+
+char _ssid[WIFI_SSID_MAX_LEN+1] = {0};
+char _password[WIFI_PASSWORD_MAX_LEN+1] = {0};
+#endif // WIFI
+
+/* \brief print_MAC - print_MAC  - helper function to print out MAC address
+ * in: network_interface - pointer to network i/f
+ *     bool log-messages   print out logs or not
+ * MAC address is print, if it can be acquired & log_messages is true.
+ *
+ */
+void print_MAC(NetworkInterface* network_interface, bool log_messages) {
+#if MBED_CONF_APP_NETWORK_INTERFACE != CELLULAR_ONBOARD
+    const char *mac_addr = network_interface->get_mac_address();
+    if (mac_addr == NULL) {
+        if (log_messages) {
+            printf("[EasyConnect] ERROR - No MAC address\n");
+        }
+        return;
+    }
+    if (log_messages) {
+        printf("[EasyConnect] MAC address %s\n", mac_addr);
+    }
+#endif
+}
+
+
+
+/* \brief easy_connect     easy_connect() function to connect the pre-defined network bearer,
+ *                         config done via mbed_app.json (see README.md for details).
+ * IN: bool  log_messages  print out diagnostics or not.
+ */
+NetworkInterface* easy_connect(bool log_messages) {
+    NetworkInterface* network_interface = NULL;
+    int connect_success = -1;
+
+#if defined (WIFI)
+    // We check if the _ssid and _password have already been set (via the easy_connect() that takes thoses parameters or not
+    // If they have not been set, use the ones we can gain from mbed_app.json.
+    if (strlen(_ssid) == 0) { 
+        if(strlen(MBED_CONF_APP_WIFI_SSID) > WIFI_SSID_MAX_LEN) {
+            printf("WARNING - MBED_CONF_APP_WIFI_SSID is too long - it will be cut to %d chars.\n", WIFI_SSID_MAX_LEN);
+        }
+        strncpy(_ssid, MBED_CONF_APP_WIFI_SSID, WIFI_SSID_MAX_LEN);
+    }
+
+    if (strlen(_password) == 0 ) {
+        if(strlen(MBED_CONF_APP_WIFI_PASSWORD) > WIFI_PASSWORD_MAX_LEN) {
+            printf("WARNING - MBED_CONF_APP_WIFI_PASSWORD is too long - it will be cut to %d chars.\n", WIFI_PASSWORD_MAX_LEN);
+        }
+        strncpy(_password, MBED_CONF_APP_WIFI_PASSWORD, WIFI_PASSWORD_MAX_LEN);
```

Comment:
Normally I'm a big fan of avoiding dynamic allocation whenever possible, but this might be a good case to use the heap? Is there another requirement for using static allocation?

Reply:
Well, me neither - most dynamic allocs will eventually fragment the heap. However, typically this one would be done just once at start... Passwords are rarely longer than say 8-14 characters, so there would be potential savings.